spec/support/factory_girl.rb

```diff
@@ -13,6 +14,7 @@
 
 FactoryGirl.define do
   factory :user do
+    id 4
```

Comment:
You don't need to specific the primary key in the factory. The Factory will create a new one for you. Remove this.